docs/modules/ROOT/pages/how-tos/exoscale/change_storage_node_size.adoc

```diff
@@ -0,0 +1,494 @@
++
+[source,bash]
+----
+terraform apply
+----
+
+. Approve node certs for new storage nodes
++
+include::partial$install/approve-node-csrs.adoc[]
+
+. Label and taint the new storage nodes
++
+include::partial$exoscale/label-taint-storage-nodes.adoc[]
+
+. Wait for the localstorage PVs on the new nodes to be created
++
+[source,bash]
+----
+kubectl --as=cluster-admin get pv \
+  -l storage.openshift.com/local-volume-owner-name=storagevolumes -w
+----
+
+. Disable auto sync for component `rook-ceph`.
+This allows us to temporarily make manual changes to the Rook Ceph cluster.
++
+include::partial$disable-argocd-autosync.adoc[]
+
+. Make a note of the original count of OSDs in the Ceph cluster
++
+[source,bash]
+----
+orig_osd_count=$(kubectl --as=cluster-admin -n syn-rook-ceph-cluster \
+  get cephcluster cluster -o jsonpath='{.spec.storage.storageClassDeviceSets[0].count}')
+echo $orig_osd_count
+----
+
+. Change Ceph cluster to lauch new OSDs
++
+[source,bash]
+----
+kubectl --as=cluster-admin -n syn-rook-ceph-cluster patch cephcluster cluster --type=json \
+  -p "[{
+    \"op\": \"replace\",
+    \"path\": \"/spec/storage/storageClassDeviceSets/0/count\",
+    \"value\": $(expr ${orig_osd_count} '*' 2)
+  }]"
+----
+
+. Wait until the new OSDs are launched
++
+[source,bash]
+----
+kubectl --as=cluster-admin -n syn-rook-ceph-cluster get pods -w
+
+# Wait for all PGs to reach `active+clean` state
+watch kubectl --as=cluster-admin -n syn-rook-ceph-cluster exec -it deploy/rook-ceph-tools -- \
+  ceph -s
+----
+
+=== Remove old OSD
+
+. Tell Ceph to take OSDs on old nodes out of service and relocate data stored on them
++
+[source,bash]
+----
+for node in $(echo -n $NODES_TO_REPLACE); do
+  osd_id=$(kubectl --as=cluster-admin -n syn-rook-ceph-cluster get deploy \
+    -l failure-domain="${node}" --no-headers \
+    -o custom-columns="NAME:.metadata.name" | cut -d- -f4)
+  echo Taking OSD out of service:
+  kubectl --as=cluster-admin -n syn-rook-ceph-cluster get pods -o wide | grep "osd-${osd_id}"
+  echo -n "Is the node correct? (y/N) "; read REPLY; echo
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+      kubectl --as=cluster-admin -n syn-rook-ceph-cluster exec -it deploy/rook-ceph-tools -- \
+        ceph osd crush reweight "osd.${osd_id}" 0
+  fi
+done
+----
+
+. Wait for the data to be redistributed ("backfilled") to the other OSDs.
++
+include::partial$storage-ceph-backfilling.adoc[]
+
+. Remove the OSDs from the Ceph cluster
++
+[source,bash]
+----
+for node in $(echo -n $NODES_TO_REPLACE); do
+  osd_id=$(kubectl --as=cluster-admin -n syn-rook-ceph-cluster get deploy \
+    -l failure-domain="${node}" --no-headers \
+    -o custom-columns="NAME:.metadata.name" | cut -d- -f4)
+
+  kubectl --as=cluster-admin -n syn-rook-ceph-cluster scale --replicas=0 \
+    "deploy/rook-ceph-osd-${osd_id}"
+  kubectl --as=cluster-admin -n syn-rook-ceph-cluster exec -it deploy/rook-ceph-tools -- \
+    ceph osd purge "${osd_id}" --yes-i-really-mean-it
```

Comment:
Is `--yes-i-really-mean-it` actually required here?

Reply:
I'll test it in the final round of testing 👍